scripts/generate-types.js

```diff
@@ -1,88 +1,128 @@
-const fs = require("fs");
-
 const { pascalCase } = require("pascal-case");
-const prettier = require("prettier");
 const TypeWriter = require("@gimenete/type-writer");
-const webhooks = require("@octokit/webhooks-definitions");
+const webhooks = require("@octokit/webhooks-definitions/index.json");
+const { generateFile } = require("./generate-file");
 
 const signatures = [];
+const eventEnums = [];
 const tw = new TypeWriter();
 
+const doNotEditThisFileDisclaimer = `
+// DO NOT EDIT THIS FILE DIRECTLY
+// make edits in scripts/generate-types.js
+`;
+const eventPayloadsVariable = "EventPayloads";
+const eventNamesVariable = "EventNames";
+
+const generatePayloadType = (typeName) => ({
+  rootTypeName: typeName,
+  namedKeyPaths: {
+    [`${typeName}.repository`]: "PayloadRepository",
+    // This prevents a naming colision between the payload of a `installation_repositories` event
+    // and the `repositories` attribute of a `installation` event
+    "WebhookPayloadInstallation.repositories":
+      "WebhookPayloadInstallation_Repositories",
+  },
+});
+
+const generateEventType = (event, typeName) => `
+  public on (
```

Comment:
We can use a Generic type to achieve this in one go. Greater flexibility to compose too. #141

Reply:
I can port the approach here if it seems okay. :)